os/generic_win_base.h

```diff
@@ -24,7 +24,10 @@
 // Unices provide `howmany` via sys/param.h
 #define howmany(x, y)  (((x) + ((y) - 1)) / (y))
 
+#ifndef _MODE_T_
+#define	_MODE_T_
```

Comment:
Is there no generic way to avoid this? I'd rather not encode the macro that MinGW uses for this into the header. Why not hoist this into a check in CMake and use `HAVE_MODE_T` instead?